File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int start=0,end=height.length-1;
        int max_area=Math.min(height[end],height[start])*(end-start);
        while (start<end){
            if(height[start]<height[end]){
                start++;
            }
            else{
                end--;
            }
            max_area=Math.max(Math.min(height[end],height[start])*(end-start),max_area);
        }
        return max_area;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        start,end=0,len(height)-1
        max_area=min(height[start],height[end])*(end-start)
        while start<end:
            if height[start]<height[end]:
                start+=1
            else:
                end-=1
            max_area=max(min(height[start],height[end])*(end-start),max_area)
        return max_area
